lib/terminal.coffee

```diff
@@ -46,9 +50,13 @@ module.exports = class Terminal extends EventEmitter
     "#{protocol}://#{@host}:#{@port}/#{@path}?token=#{@token}&version=#{version}"
 
   reset: ->
-    console.log('term:reset')
     @socket.reset()
 
+  sendWithPromise: (msg, test) ->
```

Comment:
took me a sec to grok what this function did, wondering if there's a more detailed name we can give like `sendAndReceive` or `sendAndHandleReponse` (not tied to either of those name, but something a little more specific then `sendWithPromise`